packages/pigeon/platform_tests/test_plugin/windows/test_plugin.cpp

```diff
@@ -23,24 +23,55 @@ using core_tests_pigeontest::AnEnum;
 using core_tests_pigeontest::ErrorOr;
 using core_tests_pigeontest::FlutterError;
 using core_tests_pigeontest::FlutterIntegrationCoreApi;
+using core_tests_pigeontest::FlutterSmallApi;
 using core_tests_pigeontest::HostIntegrationCoreApi;
+using core_tests_pigeontest::HostSmallApi;
 using flutter::EncodableList;
 using flutter::EncodableMap;
 using flutter::EncodableValue;
 
+// static
+void TestSmallApi::RegisterWithRegistrar(
+    flutter::PluginRegistrarWindows* registrar,
+    const std::string& message_channel_suffix) {
+  auto api = std::make_unique<TestSmallApi>(registrar->messenger());
+
+  HostSmallApi::SetUp(registrar->messenger(), api.get(),
```

Comment:
You're passing a raw pointer to `api` here, but `api` is a smart pointer whose only owner is the local scope of this function, so as soon as `RegisterWithRegistrar` returns, the handler for Pigeon calls is a dangling pointer.

You either need to make `TestSmallApi` a `Plugin` and use `AddPlugin` to keep it around, or you need to remove this whole quasi-plugin method and just have `TestPlugin` call the constructor for this object and keep it around as an ivar. I think the latter would be less confusing, since this isn't actually a plugin with a registration method that Flutter will call.

Reply:
Makes sense, I tried to keep the structure as similar to the other class as possible, but it seems it was a mistake here. Thanks